Fix rag finetuning + add finetuning test (#8585)
huggingface/transformers: examples/rag/README.md

    --print_predictions \
    --recalculate \ # adding this parameter will force recalculating predictions even if predictions_path already exists
```

# Use your own knowledge source

> RAG uses by default the enghlish wikipedia as a knowledge source, known as the 'wiki_dpr' dataset.

Reviewer: ```suggestion
By default, RAG uses the English Wikipedia as a knowledge source, known as the 'wiki_dpr' dataset.
```

> You can build your own knowledge source using the `use_custom_knowledge_dataset.py` and use it for RAG.

Reviewer: ```suggestion
With `use_custom_knowledge_dataset.py` you can build your own knowledge source, *e.g.* for RAG.
```

> Let's say you have documents in tab-separated csv files with columns "title" and "text", then you can run for example:

Reviewer: ```suggestion
For instance, if documents are serialized as tab-separated csv files with the columns "title" and "text", one can use `use_own_knowledge_dataset.py` as follows:
```
Let's say you have documents in tab-separated csv files with columns "title" and "text", then you can run for example:
```

```bash
python examples/rag/use_own_knowledge_dataset.py \
    --csv_path path/to/my_csv \
    --output_dir path/to/my_knowledge_dataset \
```

> To use it in the RAG finetuning script you can then do:

Reviewer: ```suggestion
The created outputs in `path/to/my_knowledge_dataset` can then be used to finetune RAG as follows:
```
